Extensions/Leaderboards/leaderboardstools.ts

```diff
@@ -7,10 +7,36 @@ namespace gdjs {
       let _lastPlayerName: string;
       let _lastErrorCode: number;
       let _leaderboardViewIframe: HTMLIFrameElement | null = null;
+      let _leaderboardViewIframeErrored: boolean = false;
+      let _leaderboardViewIframeLoading: boolean = false;
+      let _leaderboardViewIframeLoaded: boolean = false;
+      let _errorTimeoutId: NodeJS.Timeout | null = null;
       let _leaderboardViewClosingCallback:
         | ((event: MessageEvent) => void)
         | null = null;
 
+      const _loaderContainer: HTMLDivElement = document.createElement('div');
+      _loaderContainer.style.backgroundColor = '#000000';
+      _loaderContainer.style.display = 'flex';
+      _loaderContainer.style.height = '100%';
+      _loaderContainer.style.width = '100%';
+      _loaderContainer.style.justifyContent = 'center';
+      _loaderContainer.style.alignItems = 'center';
+      const _loader = document.createElement('img');
+      _loader.setAttribute('width', '50px');
+      _loader.setAttribute(
+        'src',
+        'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGZpbGw9Im5vbmUiIHZpZXdCb3g9IjAgMCAyNCAyNCI+CjxjaXJjbGUgb3BhY2l0eT0nMC4yNScgY3g9IjEyIiBjeT0iMTIiIHI9IjEwIiBzdHJva2U9IiNGRkZGRkYiIHN0cm9rZS13aWR0aD0iNCI+PC9jaXJjbGU+CjxwYXRoIG9wYWNpdHk9JzAuNzUnIGZpbGw9IiNGRkZGRkYiIGQ9Ik00IDEyYTggOCAwIDAxOC04VjBDNS4zNzMgMCAwIDUuMzczIDAgMTJoNHptMiA1LjI5MUE3Ljk2MiA3Ljk2MiAwIDAxNCAxMkgwYzAgMy4wNDIgMS4xMzUgNS44MjQgMyA3LjkzOGwzLTIuNjQ3eiI+PC9wYXRoPgo8L3N2Zz4='
+      );
+      _loader.animate(
```

Comment:
animate supports started with `iOS 13.1`... I think it's good but we could almost put a try/catch around it to allow this to work (without animation) without crashing the game on iOS 12/13 (even if it's <2% of iOS usage).

Reply:
Also, will this consume CPU even if not rendered? 🤔

Reply:
> animate supports started with iOS 13.1... I think it's good but we could almost put a try/catch around it to allow this to work (without animation) without crashing the game on iOS 12/13 (even if it's <2% of iOS usage).

Good idea, I'll add it

> Also, will this consume CPU even if not rendered? 🤔

Good question, I'll check

Reply:
I couldn't find something on the internet explaining how Animate would behave on a node that is not in the DOM.
I made a test on a dedicated HTML file. Here are the performance analysis by Chrome DevTools:
- First case: Loader is inserted in the div at start, then animated at 1.5s.

<img width="590" alt="image" src="https://user-images.githubusercontent.com/32449369/161012296-ac9626a1-7946-4c7e-be81-9aebfebb0d75.png">

- Second case: Loader is inserted in the div at 3s, bu still animated at 1.5s

<img width="592" alt="image" src="https://user-images.githubusercontent.com/32449369/161012369-133db40a-c661-4fb3-9f03-3d94ce3c8a12.png">

The animation seems to not require CPU when the element is not displayed: CPU line is hardly visible but:
- in the first case, we can see a little "peak" at 1.5s 
- in the second case, there seems to be a small CPU Usage when the element appears at 3s but none when the animation starts at 1.5s

Reply:
Another point is that I found a few articles claiming that infinite CSS animations can consume a lot of CPU. But the Animate API doesn't seem to inject CSS on the element.

Reply:
> The animation seems to not require CPU when the element is not displayed

Ok! I'm a bit paranoid on having any DOM work that would still the browser CPU time from the game and that would make the browser not 100% dedicated on running the game frames.